### DISCUSSION TITLE
TransactionImpl#commitTransaction not using the Gateway's commitHandler

### DISCUSSION THREAD

**Discussion by Breina** [hyperledger/fabric-gateway-java#154] [General]
When committing a transaction, the current implementation waits for no peer events at all. Despite a commit handler being specifically configured through `Gateway.Builder#commitHandler`.

The current implementation specifically comments `// Disable default commit wait behaviour`, so this behaviour seems to be intended.

I'm just wondering why this this specific strategy was chosen and if it's definitely intended.

cc: @bestbeforetoday

Code: https://github.com/hyperledger/fabric-gateway-java/blob/main/src/main/java/org/hyperledger/fabric/gateway/impl/TransactionImpl.java#L149

**Comment by bestbeforetoday** [comment]
The current implementation does wait for peer events, unless you specifically configure it not to wait for events by specifying the [NONE](https://hyperledger.github.io/fabric-gateway-java/main/org/hyperledger/fabric/gateway/DefaultCommitHandlers.html#NONE) commit handler. The commit handling behaviour of the underlying fabric-sdk-java API is disabled and instead fabric-gateway-java's own commit handling implementation is used, which provides more flexibility:

https://github.com/hyperledger/fabric-gateway-java/blob/c6429466a48ac0aa71e99807f4c3ed533c283ef6/src/main/java/org/hyperledger/fabric/gateway/impl/TransactionImpl.java#L160

**Comment by Breina** [comment]
Ah very well, thanks for the quick retort!
